linode/manager
# upcoming: [M3-7614] - Placement Groups Create/Rename Drawers

## Description 📝
This PR implements the "Create" & "Rename" Placement Groups drawers. It does a simple abstraction of its shared components to which we pass the formik props and optional placement group data.

I wanted to avoid relying on a "mode" state to have a more robust separation of concern between both drawers and avoid mashing two mutations together. It's a bit more readable and scalable that way, and doesn't part from our existing conventions.

## Changes  🔄
- Add Drawers, related components and their tests
- Change affinity type to be an enum (usually a good practice for small, static select options)
- Rename "update" nomenclature to be "rename" since this is all that drawer does (and likely will do)
- Code cleanup and tightening 

## Preview 📷
| Create Drawer  | Rename Drawer   |
| ------- | ------- |
| ![Screenshot 2024-01-29 at 10 06 01](https://github.com/linode/manager/assets/130582365/a8cad600-69e7-497c-a641-099c194d8cae) | ![Screenshot 2024-01-29 at 10 05 44](https://github.com/linode/manager/assets/130582365/a37b78e6-5771-425d-b451-2d121a733f43) |

## How to test 🧪

### Prerequisites
- Have the "Placement Group" flag and MSW turned on

### Verification steps 
- Create Drawer
  - Navigate to http://localhost:3000/placement-groups
  - Click on "Create Placement Group"
  - Confirm behavior, error handling, validation and submission
- Rename Drawer
  - Navigate to http://localhost:3000/placement-groups
  - Click on "Rename" CTA for one of the three PGs in the table
  - Confirm behavior, error handling, validation and submission

## As an Author I have considered 🤔

*Check all that apply*
